contracts/KyberIEOWrapper.sol

```diff
@@ -1,7 +1,57 @@
-pragma solidity ^0.4.0;
+pragma solidity ^0.4.23;
 
-contract KyberIEOWrapper {
-    function KyberIEOWrapper(){
 
+import './KyberIEO.sol';
+import './ERC20Interface.sol';
+
+
+interface KyberNetwork {
+    function trade(ERC20 src, uint srcAmount, ERC20 dest, address destAddress, uint maxDestAmount,
+        uint minConversionRate, address walletId) external payable returns(uint);
+    function getExpectedRate(ERC20 src, ERC20 dest, uint srcQty) external view
+        returns (uint expectedRate, uint slippageRate);
+}
+
+
+contract KyberIEOWrapper is Withdrawable {
+
+    KyberNetwork kyberNetwork;
+    KyberIEO kyberIeo;
+    ERC20 constant internal ETH_TOKEN_ADDRESS = ERC20(0x00eeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeee);
+
+    constructor(KyberNetwork _kyber, address _admin) public Withdrawable(_admin) {
+        require (_kyber != address(0));
+        kyberNetwork = _kyber;
+    }
+
+    event Contribution(address contributor, ERC20 token, uint amountTwei, uint tradedWei, uint change);
+    function contributeWithToken(
+        address contributor,
+        ERC20 token,
+        uint amountTwei,
+        uint8 v,
+        bytes32 r,
+        bytes32 s) public returns(uint)
+    {
+        uint weiCap = kyberIeo.getContributorRemainingCap(contributor);
+        require(weiCap > 0);
+
+        uint expectedRate;
+        uint slippageRate;
+        (expectedRate, slippageRate) = kyberNetwork.getExpectedRate(token, ETH_TOKEN_ADDRESS, amountTwei);
+
+        require(expectedRate > 0);
+        require(token.transferFrom(contributor, this, amountTwei));
+
+        token.approve(address(kyberNetwork), amountTwei);
```

Comment:
you forgot to reset it

Reply:
r u sure.
If I take the whole sum it zeroes. no?